ads/pubmine.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript, validateData} from '../3p/3p';
+import {getSourceOrigin, getSourceUrl} from '../src/url';
+
+const pubmineFields = ['adsafe', 'section', 'siteid', 'wordads'],
+  pubmineRequired = ['siteid'],
+  pubmineURL = 'https://s.pubmine.com/showad.js';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function pubmine(global, data) {
+  validateData(data, pubmineRequired, pubmineFields);
```

Comment:
the 2nd param of `validateData` is `optionalFields`. So you can rename the var and remove `'siteid'` from it.